File: medplum-typescript/packages/sdk/src/model/fhir/ExplanationOfBenefit_Item.ts
import { Address } from "./Address";
import { CodeableConcept } from "./CodeableConcept";
import { decimal } from "./decimal";
import { ExplanationOfBenefit_Adjudication } from "./ExplanationOfBenefit_Adjudication";
import { ExplanationOfBenefit_Detail } from "./ExplanationOfBenefit_Detail";
import { Extension } from "./Extension";
import { Money } from "./Money";
import { Period } from "./Period";
import { positiveInt } from "./positiveInt";
import { Quantity } from "./Quantity";
import { Reference } from "./Reference";

export interface ExplanationOfBenefit_Item {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    sequence: positiveInt | null | undefined;
    careTeamSequence: positiveInt[] | null | undefined;
    diagnosisSequence: positiveInt[] | null | undefined;
    procedureSequence: positiveInt[] | null | undefined;
    informationSequence: positiveInt[] | null | undefined;
    revenue: CodeableConcept | null | undefined;
    category: CodeableConcept | null | undefined;
    productOrService: CodeableConcept;
    modifier: CodeableConcept[] | null | undefined;
    programCode: CodeableConcept[] | null | undefined;
    servicedDate: string | null | undefined;
    servicedPeriod: Period | null | undefined;
    locationCodeableConcept: CodeableConcept | null | undefined;
    locationAddress: Address | null | undefined;
    locationReference: Reference | null | undefined;
    quantity: Quantity | null | undefined;
    unitPrice: Money | null | undefined;
    factor: decimal | null | undefined;
    net: Money | null | undefined;
    udi: Reference[] | null | undefined;
    bodySite: CodeableConcept | null | undefined;
    subSite: CodeableConcept[] | null | undefined;
    encounter: Reference[] | null | undefined;
    noteNumber: positiveInt[] | null | undefined;
    adjudication: ExplanationOfBenefit_Adjudication[] | null | undefined;
    detail: ExplanationOfBenefit_Detail[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/ExplanationOfBenefit_Item.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableExplanationOfBenefit_Item.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ExplanationOfBenefit_Item {
  CodeableConcept productOrService();

  Optional<Period> servicedPeriod();

  Optional<CodeableConcept> bodySite();

  Optional<List<Reference>> encounter();

  Optional<String> id();

  Optional<List<positiveInt>> procedureSequence();

  Optional<List<CodeableConcept>> subSite();

  Optional<List<positiveInt>> informationSequence();

  Optional<String> servicedDate();

  Optional<List<CodeableConcept>> programCode();

  Optional<List<CodeableConcept>> modifier();

  Optional<Address> locationAddress();

  Optional<List<positiveInt>> careTeamSequence();

  Optional<decimal> factor();

  Optional<Reference> locationReference();

  Optional<List<positiveInt>> diagnosisSequence();

  Optional<CodeableConcept> revenue();

  Optional<List<ExplanationOfBenefit_Adjudication>> adjudication();

  Optional<List<ExplanationOfBenefit_Detail>> detail();

  Optional<Money> net();

  Optional<CodeableConcept> category();

  Optional<List<positiveInt>> noteNumber();

  Optional<List<Extension>> extension();

  Optional<CodeableConcept> locationCodeableConcept();

  Optional<Money> unitPrice();

  Optional<Quantity> quantity();

  Optional<List<Extension>> modifierExtension();

  Optional<positiveInt> sequence();

  Optional<List<Reference>> udi();

  static ImmutableExplanationOfBenefit_Item.ProductOrServiceBuildStage builder() {
    return ImmutableExplanationOfBenefit_Item.builder();
  }
}
